gravity-forms/gw-field-to-field-conditional-logic.php

```diff
@@ -236,15 +236,15 @@ public function modify_rule( $rule, $form, $logic, $field_values, $entry ) {
 			$_is_modifying_rule = false;
 		}
 
-		if ( ! isset( $_rule_cache[ $entry['id'] ] ) ) {
-			$_rule_cache[ $entry['id'] ] = array();
+		if ( ! isset( $_rule_cache[ rgar( $entry, 'id' ) ] ) ) {
+			$_rule_cache[ rgar( $entry, 'id' ) ] = array();
 		}
 
-		if ( isset( $_rule_cache[ $entry['id'] ][ $rule['value'] ] ) ) {
-			$value = $_rule_cache[ $entry['id'] ][ $rule['value'] ];
+		if ( isset( $_rule_cache[ rgar( $entry, 'id' ) ][ $rule['value'] ] ) ) {
+			$value = $_rule_cache[ rgar( $entry, 'id' ) ][ $rule['value'] ];
 		} else {
 			$value = GFCommon::replace_variables( $rule['value'], $form, $entry );
-			$_rule_cache[ $entry['id'] ][ $rule['value'] ] = $value;
+			$_rule_cache[ rgar( $entry, 'id' ) ][ $rule['value'] ] = $value;
```

Comment:
With all of these calls I'd create a new `$entry_id` variable to reduce some duplication.